File: devopsfront/src/api/project.js
import request from '@/utils/request'

export function getProjects(params) {
  return request({
    url: '/api/projects',
    method: 'get',
    params
  })
}

export function getProject(id) {
  return request({
    url: `/api/projects/${id}`,
    method: 'get'
  })
}

export function createProject(data) {
  return request({
    url: '/api/projects',
    method: 'post',
    data
  })
}

export function updateProject(id, data) {
  return request({
    url: `/api/projects/${id}`,
    method: 'put',
    data
  })
}

export function deleteProject(id) {
  return request({
    url: `/api/projects/${id}`,
    method: 'delete'
  })
} 

File: devops/router/project.go
package router

import (
	"devops/controllers"
	"github.com/gin-gonic/gin"
)

// SetupProjectRoutes 设置项目路由
func SetupProjectRoutes(router *gin.RouterGroup) {
	projectController := controllers.NewProjectController()

	// 项目路由组
	projects := router.Group("/projects")
	{
		projects.POST("", projectController.CreateProject)
		projects.GET("", projectController.GetProjects)
		projects.GET("/:id", projectController.GetProject)
		projects.PUT("/:id", projectController.UpdateProject)
		projects.DELETE("/:id", projectController.DeleteProject)
	}
}
